src/Symfony/Component/PropertyInfo/Type.php

```diff
@@ -57,9 +57,12 @@ class Type
     private $collectionValueType;
 
     /**
+     * @param array|Type|null $collectionKeyType
+     * @param array|Type|null $collectionValueType
```

Comment:
why not `Type[]|Type|null`? `validateCollectionArgument` below looks like a runtime safeguard, im not sure we need.

Reply:
I've fixed the phpDoc, about `validateCollectionArgument` I tried to make something that will trigger exception as close as PHP 8 with union types, I do think it's the best in order to replace it with union types when PHP 8 will be minor version in Symfony.